resolver-dns/src/main/java/io/netty/resolver/dns/UnixResolverDnsServerAddressStreamProvider.java

```diff
@@ -167,48 +167,52 @@ private boolean mayOverrideNameServers() {
-                                    " in file " + etcResolverFile + " value: " + line);
+                            String maybeIP = line.substring(i);
+                            // There may be a port appended onto the IP address so we attempt to extract it.
+                            if (!NetUtil.isValidIpV4Address(maybeIP) && !NetUtil.isValidIpV6Address(maybeIP)) {
+                                i = maybeIP.lastIndexOf('.');
+                                if (i + 1 >= maybeIP.length()) {
+                                    throw new IllegalArgumentException("error parsing label " + NAMESERVER_ROW_LABEL +
+                                            " in file " + etcResolverFile + ". invalid IP value: " + line);
+                                }
+                                port = Integer.parseInt(maybeIP.substring(i + 1));
+                                maybeIP = maybeIP.substring(0, i);
+                            }
+                            addresses.add(SocketUtils.socketAddress(maybeIP, port));
+                        } else if (line.startsWith(DOMAIN_ROW_LABEL)) {
+                            int i = indexOfNonWhiteSpace(line, DOMAIN_ROW_LABEL.length());
+                            if (i < 0) {
+                                throw new IllegalArgumentException("error parsing label " + DOMAIN_ROW_LABEL +
+                                        " in file " + etcResolverFile + " value: " + line);
+                            }
+                            domainName = line.substring(i);
+                            if (!addresses.isEmpty()) {
+                                putIfAbsent(domainToNameServerStreamMap, domainName, addresses);
+                            }
+                            addresses = new ArrayList<InetSocketAddress>(2);
+                        } else if (line.startsWith(PORT_ROW_LABEL)) {
+                            int i = indexOfNonWhiteSpace(line, PORT_ROW_LABEL.length());
+                            if (i < 0) {
+                                throw new IllegalArgumentException("error parsing label " + PORT_ROW_LABEL +
+                                        " in file " + etcResolverFile + " value: " + line);
+                            }
+                            port = Integer.parseInt(line.substring(i));
+                        } else if (line.startsWith(SORTLIST_ROW_LABEL)) {
+                            logger.info("row type {} not supported. Ignoring line: {}", SORTLIST_ROW_LABEL, line);
                         }
-                        port = Integer.parseInt(line.substring(i));
-                    } else if (line.startsWith(SORTLIST_ROW_LABEL)) {
-                        logger.info("row type {} not supported. ignoring line: {}", SORTLIST_ROW_LABEL, line);
+                    } catch (IllegalArgumentException e) {
+                        logger.info("Could not parse entry. Ignoring line: {}", line, e);
```

Comment:
For me:

* error = blocking issue
* warn = non blocking
* info = flow progress
* debug = verbose

I'd go with warn instead of info for this.

Reply:
ok then lets use warn...